### ISSUE TITLE
Adding custom translations to code systems

### ISSUE THREAD

**Issue by albert-t25** [IHTSDO/snowstorm#637]
I have setup a Snowstorm instance and have loaded SNOMED CT International, SNOMED CT Austrian edition as well as LOINC. I'm able to read the data using FHIR API, but so far have been unable to make any kind of update. In my scenario, I would like to use the support for translations (https://build.fhir.org/languages.html#term) in order to add custom translations to both SNOMED and LOINC. However any kind of PUT/PATCH/POST request is returning a message similar to 

_Invalid request: The FHIR endpoint on this server does not know how to handle POST operation[CodeSystem] with parameters [[]]_

Are update operations on code systems supported in Snowstorm FHIR API? If so, is there some documentation for it?

Thank you

**Comment by kaicode** [comment]
[This postman collection](https://www.postman.com/interstellar-zodiac-547386/snowstorm/folder/hwetsfq/codesystem) is the currently the best documentation for the Snowstorm FHIR API capabilities.
Currently REST only supports read for the CodeSystem resource.

SNOMED CT CodeSystems must be created via SNOMED API. Other CodeSystems can be created via other package import types. It's not possible to add supplements at this time.

When a SNOMED CT edition is imported all preferred and acceptable terms are imported. These will be presented as designations when using CodeSystem/$lookup. If there are multiple dialects within the SNOMED edition one can be selected for display using the displayLanguage parameter. 
displayLanguage supports these formats:
- xx (language code only)
  - Examples: en, da, fr
  - These just use the language code in the descriptions. The others below use language and dialect.
- xx-yy (language code and dialect code)
  - Examples: en-us, en-gb, fr-be
  - These are configured here: https://github.com/IHTSDO/snowstorm/blob/10.4.2/src/main/resources/application.properties#L244
- xx-X-zzzzzz (language code and extension)
- xx-yy-X-zzzzzz (language code, dialect and extension)
- xx-X-00000000 (language code and dialect using SNOMED SCTID)
  - Example: xx-X-900000000000509007
  - This format is slightly different to the one supported in Ontoserver which uses something like xx-X-9000-0000-0000-5090-07... we have a long outstanding task to support this format too.

**Comment by albert-t25** [comment]
Thank you @kaicode. If I understood correctly, there's no way to update existing concepts in Snomed/Loinc code systems through FHIR API (or any other API for that matter), right?

**Comment by kaicode** [comment]
Correct. If you want to add terms to SNOMED CT that would have to happen via a SNOMED RF2 extension.
I don't know how to add terms to a loinc package.

Some FHIR terminology servers allow adding terms to existing codes via a CodeSystem supplement, but Snowstorm does not support that yet.

**Comment by albert-t25** [comment]
Is there any plan to support adding terms to existing codes in Snowstorm through FHIR API? If not, would you be willing to accept contributions toward that?

Thank you

**Comment by kaicode** [comment]
Adding terms to existing codes must be happen using a CodeSystem supplement.
See slides 5-14 of this DevDays presentation [Mastering FHIR Terminology (Dion McMurtrie, CSIRO)](https://www.devdays.com/wp-content/uploads/2023/08/230607_DionMcMurtrie_Mastering-FHIR-Terminology-Navigating-the-advanced-services-for-software-developers.pdf)

There is no current plan to implement this functionality. We would accept contributions that implement adding terms using that method. Although the method outlined in the slides requires the FHIR R5 parameter `useSupplement`, and Snowstorm currently implements FHIR R4, I think supporting that parameter would be the right thing to do.

We would not accept PRs that allow modifying the imported CodeSystem directly because making changes like would break the terminology ecosystem.
